llap-common/src/java/org/apache/hadoop/hive/llap/LlapUtil.java

```diff
@@ -401,4 +401,24 @@ public static Credentials credentialsFromByteArray(byte[] binaryCredentials)
     credentials.readTokenStorageStream(dib);
     return credentials;
   }
+
+  /**
+   * @return returns the value of IS_CLOUD_DEPLOYMENT from either environment variable or system properties
+   */
+  public static boolean isCloudDeployment() {
+    return "true".equalsIgnoreCase(System.getenv("IS_CLOUD_DEPLOYMENT"))
```

Comment:
alternative you can use is to default the external ports to 0. If port number is set to 0 then it is not cloud deployment.